common/thorhelper/thorsoapcall.cpp

```diff
@@ -170,6 +170,23 @@ class Url : public CInterface, implements IInterface
         return;
     }
 
+    void cleanPath()
+    {
+        int len = path.length();
+        if (len <= 1)
+            return;
+        int gap = 0;
+        for (int i = 1; i < len; i++)
+        {
+            if (path.charAt(i) == '/' && path.charAt(i-1-gap) == '/')
+                gap++;
+            else if (gap > 0)
+                path.setCharAt(i-gap, path.charAt(i));
```

Comment:
may not be critical, but calling setCharAt per character is not particularly efficient.
If expected to be hit a lot, would be better to deal with underlying string and reset 'path' to new string when done.
e.g. could detach() and work on const char *writeptr